sample_project/Assets/SampleViewer/Samples/TimeOfDay/Scripts/SkyControllerComponent.cs

```diff
@@ -0,0 +1,359 @@
+                animateToggle.interactable = false;
+                timeSlider.interactable = false;
+            }
+            else if (skyMode == SkyMode.Animated)
+            {
+                timeSlider.value = (float)time;
+                simulateToggle.interactable = true;
+                startButton.GetComponentInChildren<TextMeshProUGUI>().text = "Start";
+                skyMode = SkyMode.None;
+                timeSlider.interactable = true;
+            }
+            else if (skyMode == SkyMode.Simulated)
+            {
+                timeSlider.value = (float)time;
+                animateToggle.interactable = true;
+                startButton.GetComponentInChildren<TextMeshProUGUI>().text = "Start";
+                skyMode = SkyMode.None;
+                timeSlider.interactable = true;
+            }
+
+            ChangeIconColor();
+        });
+
+        startTimeIncrease.onClick.AddListener(delegate
+        {
+            if (startTime < 24)
+            {
+                startTime += 1.0;
+            }
+
+            CalculateTime(startTime, startTimeText);
+        });
+
+        startTimeDecrease.onClick.AddListener(delegate
+        {
+            if (startTime > 0)
+            {
+                startTime -= 1.0;
+            }
+
+            CalculateTime(startTime, startTimeText);
+        });
+
+        stopTimeIncrease.onClick.AddListener(delegate
+        {
+            if (stopTime < 24)
+            {
+                stopTime += 1.0;
+            }
+
+            CalculateTime(stopTime, stopTimeText);
+        });
+
+        stopTimeDecrease.onClick.AddListener(delegate
+        {
+            if (stopTime > 0)
+            {
+                stopTime -= 1.0;
+            }
+
+            CalculateTime(stopTime, stopTimeText);
+        });
+
+        speedSlider.onValueChanged.AddListener(delegate
+        {
+            speed = speedSlider.value / 100;
+            var speedValue = speed * 100;
+            speedText.text = String.Format("{0:0.#}", speedValue);
+        });
+
+        collapseButton.onClick.AddListener(delegate
+        {
+            backgroundImage.gameObject.SetActive(false);
+            settingsButton.gameObject.SetActive(true);
+        });
+
+        settingsButton.onClick.AddListener(delegate
+        {
+            backgroundImage.gameObject.SetActive(true);
+            settingsButton.gameObject.SetActive(false);
+        });
+    }
+
+    private void Update()
+    {
+        CalculateTime(time, timeText);
+
+        switch (skyMode)
+        {
+            case SkyMode.None:
+                {
+                    break;
+                }
+            case SkyMode.Animated:
+                {
+                    time += speed;
+
+                    if (time >= 24.0)
+                    {
+                        time = 0.0f;
+                    }
+                    else
+                    {
+                        if (time >= stopTime)
```

Comment:
This prevents over-night animation (start time larger than stop time). Should we make that possible?
If not, should we prevent setting the start time later than stop time (e.g., force `stop >=  start time +1` )?